tests/Types.hs

```diff
@@ -89,29 +129,156 @@ u = Unboxed 1
             1#
             1##
 
+c :: Const Int a
+c = Const 1
+
+z :: Ap Maybe Int
+z = Ap (Just 3)
+
+w :: Tree Int
+w = Node 3 [Node 4 [], Node 5 []]
+
+#if MIN_VERSION_base(4,5,0)
+data Unit' = Unit'
```

Comment:
It would be helpful to leave a comment here explaining why we have copies of all of these data types—namely, to test them with `genericLift`-based `Lift` instances, rather than `genericLiftWithPkg`-based ones.